### ISSUE TITLE
Update Initial Distribution Process on UC Folder

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#12631]

**Comment by TracyLinder** [comment]
### Sync w/ Nate

- This will be assigned to Priya
- I scheduled a sync w/ Priya & Nate & Vish to discuss the build - 6/23/23

**Comment by TracyLinder** [comment]
### Builder Huddle

- This folder process is being used across multiple folders; this update would affect the other folders as well.

**Comment by TracyLinder** [comment]
### Mtg Scheduled for Clarifications
- 6/30/23

**Comment by TracyLinder** [comment]
Q: Should these attempts close any open attempts from the previous review cycle?
- The permit case manager process 'rejected' closes out open reviews. But just in case this process isn't followed, we can add the functionality here. **Yes, close them.**


How does the current Initial Distribution process get closed? - However the review completion happens, it should close this process. (Kyle is researching)
- It doesn't seem to be closing, not sure. They use it over and over, and it always works. 
- There's no to-end date. 


Which process group?
- Intake


We need a list of the franchise processes & their codes.
- Kyle will work on this. 

Does the process need any start/end date functionality?
- No, can't think of a reason.
- Ask Pepper during DEV demo


Jaclyn asked about **automated emails**
- The customer needs to know that the review cycle has started, which reviews were inserted, and the review end date.
- All the attempts need an email to the customer. However, initial is already getting an email from the Permit Completeness Check process. This ticket will focus on the 3 redistribution attempts. 
Redistribute All - AULCC MEETING REQUIRED - The email needs to include the meeting requirement & date.
Redistribute No Franchise - NO AULCC MEETING - The email needs to inform that the customer isn't required to attend the AULCC meeting. 
 Redistribute Rejected - NO AULCC MEETING - The email needs to inform that the customer isn't required to attend the AULCC meeting. 
- [ ] Jaclyn will send Tracy the excel doc she uses to track the system emails. 💜

**Comment by TracyLinder** [comment]
Question: Should redistribution processes not save unless an initial has been saved already/previously?

**Comment by TracyLinder** [comment]
Vish is out until 7/6, so I'll need to either put the email content together myself, or ask Jaclyn or Kyle.

**Comment by KyleErich** [comment]
List of Franchise Utility processes & their codes in the UC folder:
•	51139 Alpheus (El Paso Global)
•	51142 AT&T (SWBT)
•	51143 AT&T Legacy
•	51144 AT&T Metro
•	51146 CenturyLink (Qwest)
•	51147 CyrusOne
•	51148 Chevron Pipeline
•	51149 Blue Bonnet Electric
•	51172 Crown Castle (NextG)
•	51183 FiberLight
•	51184 FiberNet
•	51214 Kinder Morgan Pipeline
•	51216 Grande Communications
•	51217 Google Fiber Texas
•	51218 Level3 Communications
•	51219 LCRA
•	51248 ONCOR
•	51249 Pedernales Electric
•	51279 Sprint
•	51306 TelePacific (Telwest)
•	51307 Time Warner Cable
•	51308 TW Telecom
•	51309 Texas Gas Service
•	51336 XO Communications
•	51337 Verizon (MCI)
•	51338 Zayo
•	51339 ZNET (Duct system)

**Comment by TracyLinder** [comment]
Q: Should these attempts close any open attempts from the previous review cycle?
- No, the reviewer needs to complete all reviews, even if the cycles start piling up.

**Comment by TracyLinder** [comment]
Need to have one more AC review meeting: Pepper, Jaclyn, Kyle, Homer, Vish, Subha
- [ ] Tracy will schedule

**Comment by TracyLinder** [comment]
## Final AC Review 8/8/23

**Agenda**
Review the criteria in the issue description & make updates
Be sure Sree gives her tech perspective
Ask if we can close #12524 

### Notes

**Comment by TracyLinder** [comment]
### Still need to scope automated emails
`The customer needs to know that the review cycle has started, which reviews were inserted, and the review end date.
`
All the attempts need an email to the customer. However, initial is already getting an email from the Permit Completeness Check process. This ticket will focus on the 3 redistribution attempts.

- Redistribute All - AULCC MEETING REQUIRED - The email needs to include the meeting requirement & date.
- Redistribute No Franchise - NO AULCC MEETING - The email needs to inform that the customer isn't required to attend the AULCC meeting.
- Redistribute Rejected - NO AULCC MEETING - The email needs to inform that the customer isn't required to attend the AULCC meeting.

**Comment by TracyLinder** [comment]
### Scheduled meeting: UC Distribution Updates - Emails & Legacy 
- 8/14/23

**Comment by TracyLinder** [comment]
### Scoping 8/14/23

- Jaclyn sent me a list of the automated emails and when they go out. Pepper would like to validate that it's still accurate. 



**Attempt Names & Email Info:**


Initial Distribution - always has a meeting


Initial Distribution No Franchise - No meeting


Redistribute All - always has a meeting

Redistribute No Franchise - No meeting

Redistribute Rejected - No meeting



**Content**



**Has Meeting**
Your Utility Coordination (UC) submittal has been distributed for review. You can view the assigned reviewers and due dates on the AB+C Portal. The list of reviewers is at the bottom of the page in the Permit Workflow section


**No Meeting**
Your Utility Coordination (UC) submittal has been distributed for review. You can view the assigned reviewers and due dates on the AB+C Portal. The list of reviewers is at the bottom of the page in the Permit Workflow section

**Comment by TracyLinder** [comment]
Hello:

Thank you for your submittal to the Austin Utility Location and Coordination Committee (AULCC). Your Utility Coordination (UC) submittal has been distributed for review. You can view the assigned reviewers and due dates on the AB+C Portal. The list of reviewers is at the bottom of the page in the Permit Workflow section.

Your AULCC Meeting date is: 24-MAY-23

Meeting Location: **Meeting Online via Microsoft Teams. You will receive an email invitation with the Teams link and optional call-in number at a later date.** Meeting Starts: 2pm

Please review your information for completeness and correctness. If any data needs to be revised, added, or deleted please //// the SmartSheet.
`SMART SHEET NEEDS TO BE UPDATED BEFORE THIS EMAIL IS LIVE` 


~contact: [Jaclyn.Lozano@austintexas.gov](mailto:Jaclyn.Lozano@austintexas.gov) and [AULCC@austintexas.gov](mailto:AULCC@austintexas.gov)~


 

AULCC Tracking Number #:UCC-230105-04-04

ROWID: 13047379

Project Name:STX-TFC Capitol Complex Phase 2-Site Utilities

Applicant: Peta-anne Donn

 

Project Contact: Peta-anne Donn

 

 

Project Description: The applicant is proposing the abandonment and removal of existing utility mains, as well as construction of new utility mains within the Right of Way of 15th Street between Colorado Street and Congress Avenue, 16th Street between Lavaca Street and Congress Avenue, and Lavaca Street, Colorado Street and Congress Avenue between 15th and 16th Streets for the TFC Capitol Complex Phase 2 project. All abandoned and proposed utilities are approximate.


 

1.  Once the Project is reviewed, please begin a Spreadsheet or Minutes Sheet to record resposes, identify areas of conflict, and note proposed resolutions.

-   Projects that have been through the AULCC previously, must contain all utility comments from previous reviews. The Minutes tab, or a Conflict/Resolution spreadsheet should be utilized for this effort.

-  Final review and clearance will be contingent on final review of the Conflict/Resolution Data and concurrence form the Project Engineer, Project Manager, Project Owner, and the AULCC Chairman.

2. A Final AULCC Clearance Letter will be granted to the Project once all the conflicts have been identified, and resolution has been verified. This letter will be required for General Permit, or Site Plan submittal to the Development Services Department Intake office on the 4th floor of One Texas Center.

CAUTION: This email was received at the City of Austin, from an EXTERNAL source. Please use caution when clicking links or opening attachments. If you believe this to be a malicious and/or phishing email, please forward this email to [cybersecurity@austintexas.gov](mailto:cybersecurity@austintexas.gov).

**Comment by TracyLinder** [comment]
### Next steps as of 8/14/23

1. Pepper & Jaclyn meeting on Friday 8/18 to iron out the email content
2. Subha can begin working on this. 
3. We'll circle back with stakeholders to finish the AC for the emails. 

@vschandramouli could you take lead on finishing the email content work with Greg & Jaclyn please? I think another 30 minutes with them next week should do.

**Comment by TracyLinder** [comment]
### Builder Huddle 8/15/23

- Create another issue for the email features
- Create another issue for updating the franchise processes
- Subha thinks AC on this ticket is good for now, she will let us know when she has questions
- Do a weekly 30 minute retro on this after Subha starts
- Subha will be out for 6 working days around 8/28. She's hoping to get the config done before she leaves. 
- Schedule initial demo for next Friday 8/25

**Comment by TracyLinder** [comment]
### Initial demo scheduled for Monday 8/28

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder Here is the demo for this ticket. Please review and let me know your feedback. Thank you.
In 1st part, I covered, Initial Distribution, Initial Distribution No Franchise attempts. I tried Redistribution also but then I went for 2nd part. So last 30 seconds can be skipped in this part.
In 2nd part, I have covered Redistribution All, Redistribution No Franchise and Redistribute Rejected attempts.

https://github.com/cityofaustin/atd-data-tech/assets/51680715/31116248-9a83-4d3e-819f-6517dd52fb74

https://github.com/cityofaustin/atd-data-tech/assets/51680715/cca5e464-575a-4a4f-bac1-6644a0ca2a59

**Comment by TracyLinder** [comment]
### Demo Notes

- Put the review date validation on the process info instead of the process attempt.

**Comment by TracyLinder** [comment]
### Stakeholder / Builder Sync 9/6/23

- Subha understands the 4 requested updates from the notes above.
- Subha will do updates and record demo for us to watch so we don't have to watch the system spin (local is extremely slow)
- Tracy will schedule next team review for after 9/11/23

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder Here is the demo attached for the feedback points given above in the last week call. Please review.

https://github.com/cityofaustin/atd-data-tech/assets/51680715/625083dc-db4e-4eae-8ef0-de466fe1553d

**Comment by Nadin-Nader** [comment]
@SubhaShroff I have sent this to the DevOps CM Team to be deployed to DEV under **SCTASK0355146**

**Comment by TracyLinder** [comment]
### Stakeholder demo 9/12/23
- Demo looks great
- This is on it's way to DEV
- Subha will test in DEV with us

**Comment by TracyLinder** [comment]
### Group UAT scheduled for Friday 9/22

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Completed the testing on DEV I found couple of things not right, I will need to fix those before sending it to TEST:
WAA accepted attempt didn't add the UC distribution- instead added Initial Distribution.
UC distribution is assigned to **AW SER Application** Discipline instead of Intake Staff unlike Initial Distribution Process.

**Comment by SubhaShroff** [comment]
[12631- 42f9193099f0 - DEV Deployment Plan.docx](https://github.com/cityofaustin/atd-data-tech/files/12625060/12631-.42f9193099f0.-.DEV.Deployment.Plan.docx)
@Nadin-Nader  Please raise a DEV deployment request for 42f9193099f0 on my behalf. Thank you.

**Comment by Nadin-Nader** [comment]
@SubhaShroff this additional commit has been sent to the DevOps Team to be deployed to DEV under **SCTASK0356206**

**Comment by SubhaShroff** [comment]
Thank you @Nadin-Nader. SCTASK0356206 is served and I tested it on DEV. It works as expected. I will prepare TEST deployment plan and give it to you for promotion request. Thanks.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Thank you for raising the request. Its deployed in TEST and I tested it. RSN - 13201179.

A question/observation:
We have a validation added for all redistribute attempts - to close the all OPEN review processes. Does 'Application Review' process counts in these review processes as its under same process group?
The validation I added counts this process also to be closed before adding any redistribute attempts.

![image](https://github.com/cityofaustin/atd-data-tech/assets/51680715/4a7aa68b-3f33-410c-bd49-e7dd0ad8b526)

**Comment by Nadin-Nader** [comment]
Sent a message to Kyle asking to UAT this on TEST. Waiting on response

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been sent to the DevOps CM Team to be deployed to DEV/TEST UNDER **SCTASK0358592**

**Comment by SubhaShroff** [comment]
Thank you @Nadin-Nader.  The feedback fix is tested in TEST after deployment. Its all set for UAT in TEST AMANDA.

**Comment by Nadin-Nader** [comment]
Update:

Its ready on TEST. Demo also approved by Jaclyn. The plan was to promote to PROD along with emails tickets.. So waiting on that. If its ok to send to PROD

**Comment by Nadin-Nader** [comment]
Waiting on emails from UC Team

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @Lpriya-LIT  I noticed recently, for UC distribution process, the attempts are removed accidentally by other's commit - e72965c6a23d9ef62a506de747267179619d10d9 (Sree's). Have requested her to correct it and make another commit. Thanks.

**Comment by Lpriya-LIT** [comment]
@SubhaShroff Created DEv/Test deployment request , SCTASK0380376

**Comment by SubhaShroff** [comment]
Thank you @Lpriya-LIT. The attempts are back on process in both DEV/TEST. Now we can send this ticket along with 13395.

**Comment by SubhaShroff** [comment]
> ### UC Team Sync on 9/25/23
> * The automated emails are in draft currently, and UC team **doesn't want this ticket to go to PROD** without the automated emails on [UC Distribution Process Emails (11/29 Sprint) (SS) (Deploying to PROD) #13395](https://github.com/cityofaustin/atd-data-tech/issues/13395)
> * I explained the existing UC folders should be tested to be sure nothing breaks. Also try redefaulting the folder.

@Nadin-Nader  @TracyLinder  Since 13395 is ready to go to PROD, can we send this ticket as well ?

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been sent to the DevOps CM Team to be deployed to PROD under **SCTASK0392131**

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been deployed to PROD with #13395, i will let Kyle and Jaclyn know

**Comment by TracyLinder** [comment]
@SubhaShroff @Nadin-Nader there is a bug report with this promote. Can you please take a look? 

![image](https://github.com/cityofaustin/atd-data-tech/assets/49930868/5e03c72b-d6a0-403e-ab34-0dc244685a4e)

**Comment by SubhaShroff** [comment]
Thank you @Nadin-Nader.  I tested it on DEV. Its not adding any reviewers to Franchise review processes. Please deploy this in TEST.
[12631- f039fb53bf66a4a863a1b341a7077256b9d99fbf- TEST Deployment Plan.docx](https://github.com/cityofaustin/atd-data-tech/files/14215613/12631-.f039fb53bf66a4a863a1b341a7077256b9d99fbf-.TEST.Deployment.Plan.docx)

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been sent to the DevOps CM Team to be deployed to TEST under **SCTASK0394159**

**Comment by SubhaShroff** [comment]
@Nadin-Nader The fix is in TEST and tested. Please review and let me know if we can send this to PROD. Thanks.
RSN - 13206729. 
![image](https://github.com/cityofaustin/atd-data-tech/assets/51680715/33f1d56f-4ddf-44cf-8957-0c1936e34542)

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been sent to the DevOps CM Team to be deployed to PROD under **SCTASK0394355**

**Comment by Nadin-Nader** [comment]
Still waiting on CTM DevOps

**Comment by SubhaShroff** [comment]
@Nadin-Nader  The fix is deployed in PROD and I have verified it for the code changes on PROD github link. The next UC permit that adds Franchise reviews should not have any assigned reviewer.  Please UAT.
Thanks.

**Comment by KyleErich** [comment]
@Nadin-Nader @SubhaShroff I'm approving on PROD since I haven't become aware of any other negative feedback about the changes. We can go ahead and close out this card and if something comes up we'll submit a new card.
